### PR TITLE
add first steps guide and more template instructions

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@CowboysFan @adellape here's the PR for the first steps guide and some related info, ptal.

File: dev_guide/templates.adoc [hunk 1]
```diff
@@ -212,3 +212,8 @@ database name                                root
 
 The output identifies several parameters that are generated with a regex
 expression generator when the template is processed.
+
+== Using the QuickStart Templates
+OpenShift provides a number of templates in the box to make it easy to quickly get started creating a new application for different languages.  Templates are provided for Rails(Ruby), Django(Python), NodeJS, CakePHP(PHP), and Dancer(Perl).  Your administrator should have registered these templates in the `openshift` project so you have access to them.  If not, direct him or her to the link:../admin_guide/install/first_steps.html[first steps] guide.
```
Comment (bparees) [inline dev_guide/templates.adoc]: @rhcarvalho @sspeiche @jhadvig @ewolinetz this is how i suggest we handle explaining to users how to use the templates.  You can link to this doc/section from the welcome page if you want, but let's not provide all this material in every repo.

(and we can improve this section of course)

File: admin_guide/install/first_steps.adoc [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+*Note:* if you are setting up a subscribed system and prefer to use the RHEL7 based images,  run this command instead:
+    
+    $ oc create -f examples/image-streams/image-streams-rhel7.json -n openshift
+
+It is not possible to create both sets of ImageStreams because they use the same name.  If you desite to have both sets of ImageStreams available to users, either create one set in a different Project, or edit one of the files and modify the ImageStream names to make them unique.
+
+== Create ImageStreams for JBoss Middleware Images
+The JBoss Middleware image streams provide images for JBoss EAP, JBoss EWS, and JBoss AMQ.  They can be used to build applications for those platforms or run directly as is.
+
+    $ oc create -f examples/middleware-streams/jboss-image-streams.json
```
Comment (bparees) [inline admin_guide/install/first_steps.adoc]: @sdodson @brenton i took a guess at the path we'll make this available under...  can we either use this path, or we need to update this doc to the actual location.

File: admin_guide/install/first_steps.adoc [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+* Service to provide load balancing for the application Pods
+* Route to provide external access to the application
+
+Some of the templates also define a database deployment and service so the application can perform database operations.
+
+Once registered, users will be able to easily instantiate full applications using the various language images provided with OpenShift.  They can also customize the template parameters during instantiation so that it build source from their own repository rather than the sample repository, so this provides a simple starting point for building new applications.
+
+To register the basic templates run:
+
+    $ oc create -n openshift -f examples/quickstart-templates
```
Comment (bparees) [inline admin_guide/install/first_steps.adoc]: @sdodson @brenton i took a guess at the path we'll make this available under...  can we either use this path, or we need to update this doc to the actual location.

File: admin_guide/install/first_steps.adoc [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+There is also a set of templates for creating applications using various JBoss products (EAP, EWS, AMQ) which can be registered by running: 
+
+    $ oc create -n openshift -f examples/middleware-templates
```
Comment (bparees) [inline admin_guide/install/first_steps.adoc]: @sdodson @brenton i took a guess at the path we'll make this available under...  can we either use this path, or we need to update this doc to the actual location.

File: admin_guide/install/first_steps.adoc [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+To create the core set of image streams, run:
+
+    $ oc create -f examples/image-streams/image-streams-centos7.json -n openshift
+
+*Note:* if you are setting up a subscribed system and prefer to use the RHEL7 based images,  run this command instead:
+    
+    $ oc create -f examples/image-streams/image-streams-rhel7.json -n openshift
+
+It is not possible to create both sets of ImageStreams because they use the same name.  If you desite to have both sets of ImageStreams available to users, either create one set in a different Project, or edit one of the files and modify the ImageStream names to make them unique.
```
Comment (adellape) [inline admin_guide/install/first_steps.adoc]: s/desite/desire/ ?

File: admin_guide/install/first_steps.adoc [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+Once registered, users will be able to easily instantiate the various templates, giving them quick access to a database deployment.
+
+To register the core set of database templates, run:
+
+    $ oc create -f examples/db-templates -n openshift
+
+
+== Create QuickStart Templates
+The QuickStart templates define a full set of components for a running application.  These include:
+* BuildConfig to build the application from source located in a GitHub public repository
```
Comment (adellape) [inline admin_guide/install/first_steps.adoc]: Needs a blank line before the start of the list.

File: admin_guide/install/first_steps.adoc (hunk 5 above)
Comment (bparees) [inline admin_guide/install/first_steps.adoc]: thanks, fixed.

File: admin_guide/install/first_steps.adoc (hunk 6 above)
Comment (bparees) [inline admin_guide/install/first_steps.adoc]: thanks, fixed.

File: admin_guide/install/first_steps.adoc [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+* DeploymentConfig to deploy the application image once built
+* Service to provide load balancing for the application Pods
+* Route to provide external access to the application
+
+Some of the templates also define a database deployment and service so the application can perform database operations.
+
+Once registered, users will be able to easily instantiate full applications using the various language images provided with OpenShift.  They can also customize the template parameters during instantiation so that it build source from their own repository rather than the sample repository, so this provides a simple starting point for building new applications.
```
Comment (adellape) [inline admin_guide/install/first_steps.adoc]: Accidentally'd a word/letter, I think:

"so that it build source"

**Comment by adellape** [comment]
LGTM, I can follow up with any minor formatting/edits.

File: dev_guide/templates.adoc (hunk 1 above)
Comment (jhadvig) [inline dev_guide/templates.adoc]: +1 for putting link for this docs into welcome page

Comment (rhcarvalho) [inline dev_guide/templates.adoc]: Will be linking to http://docs.openshift.org/latest/admin_guide/install/first_steps.html

Comment (rhcarvalho) [inline dev_guide/templates.adoc]: Actually, should be http://docs.openshift.org/latest/dev_guide/templates.html#using-the-quickstart-templates

File: admin_guide/install/first_steps.adoc (hunk 2 above)
Comment (sdodson) [inline admin_guide/install/first_steps.adoc]: Actual base path would be /usr/share/openshift/examples but that's pretty long, we could tell them to cd /usr/share/openshift first and then keep the commands as is

+1 on examples/image-streams and examples/templates
Any reason we'd want to keep middleware-streams and db-templates separate, couldn't they all go into the same directories provided there's no filename conflicts?

**Comment by bparees** [comment]
@adellape missing letter located and returned to its home.

**Comment by bparees** [comment]
merge at will.

**Comment by adellape** [comment]
:+1:

**Comment by adellape** [comment]
:shipit:

**Pull request merged**